ruby/runtime/BUILD.bazel

```diff
@@ -0,0 +1,157 @@
+load("@bazel_skylib//lib:selects.bzl", "selects")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load("@rules_ruby//ruby/private:runtime_alias.bzl",
+    _ruby_headers_alias = "ruby_headers_alias",
+    _ruby_interpreter_alias = "ruby_interpreter_alias",
+    _ruby_runtime_alias = "ruby_runtime_alias",
+)
+
+package(default_visibility = ["//visibility:public"])
+
+toolchain_type(name = "toolchain_type")
+
+# Alias targets corresponding to whichever toolchain was resolved based on
+# the selected version.
+
+_ruby_runtime_alias(
+    name = "runtime",
+)
+
+_ruby_headers_alias(
+    name = "headers",
+    runtime = ":runtime",
+)
+
+_ruby_interpreter_alias(
+    name = "interpreter",
+    runtime = ":runtime",
+)
+
+# Supported ruby versions, which can be selected by flags.
+# For example: --@rules_ruby//ruby/runtime:version=jruby-9.3
+string_flag(
+    name = "version",
+    build_setting_default = "system",
+    values = [
+        "system",
+        "ruby-2.5",
+        "ruby-2.6",
+        "ruby-2.7",
+        "ruby-3.0",
+        "ruby-3.1",
+        "jruby-9.2",
+        "jruby-9.3",
+    ],
+)
+
+config_setting(
+    name = "config_system",
+    flag_values = { "version" : "system" },
+)
+
+config_setting(
+    name = "config_ruby-2.5.8",
+    flag_values = { "version" : "ruby-2.5" },
+)
+
+alias(
+    name = "config_ruby_25",
+    actual = ":config_ruby-2.5.8"
+)
+
+config_setting(
+    name = "config_ruby-2.6.8",
+    flag_values = { "version" : "ruby-2.6" },
+)
+
+alias(
+    name = "config_ruby_26",
+    actual = ":config_ruby-2.6.8"
+)
+
+config_setting(
+    name = "config_ruby-2.7.5",
+    flag_values = { "version" : "ruby-2.7" },
+)
+
+alias(
+    name = "config_ruby_27",
+    actual = ":config_ruby-2.7.5"
+)
+
+config_setting(
+    name = "config_ruby-3.0.3",
+    flag_values = { "version" : "ruby-3.0" },
+)
+
+alias(
+    name = "config_ruby_30",
+    actual = ":config_ruby-3.0.3"
+)
+
+config_setting(
+    name = "config_ruby-3.1.1",
+    flag_values = { "version" : "ruby-3.1" },
+)
+
+alias(
+    name = "config_ruby_31",
+    actual = ":config_ruby-3.1.1"
+)
+
+selects.config_setting_group(
+    name = "config_system_ruby",
+    match_all = [
+        ":config_system",
+        "//external:rules_ruby_system_jruby_implementation",
+    ],
+)
+
+selects.config_setting_group(
+    name = "config_ruby",
+    match_any = [
+        ":config_system_ruby",
+        ":config_ruby_25",
+        ":config_ruby_26",
+        ":config_ruby_27",
+        ":config_ruby_30",
+        ":config_ruby_31",
+    ],
+)
+
+config_setting(
+    name = "config_jruby-9.2.21.0",
```

Comment:
I'm concerned about the number of different places we have to add / update ruby interpreter version numbers across this repo and the main repo. It seems like we're creating a maintenance headache. Is there anything we could do to refactor that would allow us to consolidate?

Reply:
How often do you think we'll need to update these?

With my current setup there are only 2 places that independently need to know the full version numbers (constants.bzl and runtime/BUILD.bazel).  These both list the full versions supported by rules_ruby, and map them to shorter identifiers (e.g. 3.1.1->3.1).  I think I could consolidate these with Bazel macros though, and make only a single source-of-truth.

Where in the protobuf repo do we duplicate these?  The only thing that comes to mind is our kokoro builds, but the versions we test against will always be a smaller set than those which we support.

Reply:
I consolidated this into a single source of truth, and made this file derive from that static list.  I think if we ever need to loop over versions in protobuf we can reuse these macros, but I don't think we do today